yt_dlp/extractor/acfun.py

```diff
@@ -0,0 +1,194 @@
+                'tbr': float_or_none(video.get('avgBitrate')),
+                'ext': ext
+            })
+
+        self._sort_formats(formats)
+
+        video_list = json_all['videoList']
+        p_idx, video_info = [(idx + 1, v) for (idx, v) in enumerate(video_list)
+                             if v['id'] == video_inner_id
+                             ][0]
+
+        title = json_all['title']
+        if len(video_list) > 1:
+            title = f"{title} P{p_idx:02d} {video_info['title']}"
+
+        return {
+            'id': video_id,
+            'title': title,
+            'duration': float_or_none(video_info.get('durationMillis'), 1000),
+            'timestamp': int_or_none(video_info.get('uploadTime'), 1000),
+            'formats': formats,
+            'http_headers': {
+                'Referer': url,
+            },
+        }
+
+
+class AcFunBangumiIE(InfoExtractor):
+    IE_NAME = 'AcFunBangumiIE'
+    _VALID_URL = r'''(?x)
+                    https?://
+                        (?:
+                            www\.acfun\.cn/bangumi/
+                        )
+                        (?P<id>aa[_\d]+)
+                        (?:\?ac=(?P<ac_idx>\d+))?
+                    '''
+
+    _TESTS = [{
+        'url': 'https://www.acfun.cn/bangumi/aa6002917',
+        'info_dict': {
+            'id': 'aa6002917',
+            'title': '租借女友 第1话 租借女友',
+            'duration': 1467,
+            'timestamp': 1594432800,
+        },
+        'params': {
+            # m3u8 download
+            'skip_download': True,
+        },
+        'skip': 'Geo-restricted to China',
+    }, {
+        'url': 'https://www.acfun.cn/bangumi/aa6002917',
+        'only_matching': True,
+    }, {
+        'url': 'https://www.acfun.cn/bangumi/aa6002917_36188_1745457?ac=2',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        mobj = self._match_valid_url(url)
+        video_id = mobj.group('id')
+
+        webpage = self._download_webpage(url, video_id)
+        json_bangumi_data = self._search_json(r'window.bangumiData\s*=\s*', webpage, 'bangumiData', video_id)
+
+        if not mobj.group('ac_idx'):
+            video_info = json_bangumi_data['currentVideoInfo']
+            playjson = self._parse_json(video_info['ksPlayJson'], video_id)
+            title = json_bangumi_data['showTitle']
+        else:
+            # if has ac_idx, this url is a proxy to other video which is at https://www.acfun.cn/v/ac
+            # the normal video_id is not in json
+            ac_idx = mobj.group('ac_idx')
+            video_id = f"{video_id}_ac={ac_idx}"
+            video_info = json_bangumi_data['hlVideoInfo']
+            playjson = self._parse_json(video_info['ksPlayJson'], video_id)
+            title = video_info['title']
+
+        try:
+            ext = Path(video_info['fileName']).suffix[1:]
+        except Exception:
+            self.report_warning('Parse ext failed, use mp4')
+            ext = 'mp4'
```

Comment:
Same as [this](https://github.com/yt-dlp/yt-dlp/pull/4228/commits/a338aec477b52a80d11573dd306d63bea2d8a883#r910782573)